Cesium3DTilesSelection/src/Tileset.cpp

```diff
@@ -89,41 +94,61 @@ Tileset::Tileset(
                     options.credit.value(),
                     options.showCreditsOnScreen))
               : std::nullopt),
-      _ionAssetID(ionAssetID),
-      _ionAccessToken(ionAccessToken),
-      _isRefreshingIonToken(false),
-      _ionAssetEndpointUrl(ionAssetEndpointUrl),
       _options(options),
       _pRootTile(),
       _previousFrameNumber(0),
-      _loadsInProgress(0),
-      _subtreeLoadsInProgress(0),
       _overlays(*this),
-      _tileDataBytes(0),
-      _supportsRasterOverlays(false),
       _gltfUpAxis(CesiumGeometry::Axis::Y),
       _distancesStack(),
       _nextDistancesVector(0) {
   if (ionAssetID > 0) {
     CESIUM_TRACE_USE_TRACK_SET(this->_loadingSlots);
-    this->notifyTileStartLoading(nullptr);
-    LoadIonAssetEndpoint::start(*this).thenInMainThread(
-        [this]() { this->notifyTileDoneLoading(nullptr); });
+    auto authorizationChangeListener = [this](
+                                           const std::string& header,
+                                           const std::string& headerValue) {
```

Comment:
I just realize that if the loader owns its own request headers, in the case where we have a lot of external tilesets (which we will have that much child loaders), it can become quite expensive since each child loader will need to allocate and copy the request headers from the parent loader. I think I've seen at least one tileset that has a lot external tilesets where each external covers up to 4 levels tiles. Also changing one header from the parent loader will also mean that we need to propagate that change to all the loaders in the tree, which maybe expensive as well (though such operation is done rarely). Do you have any opinions on this?

Reply:
FWIW, I think main was already copying the headers for each external tileset, so I'm not too fussed if this branch continues to do so. But if you think the current design is better than the one I proposed because of this, I won't argue with that.